be/src/storage/rowset/segment_v2/data_decoder.cpp

```diff
@@ -0,0 +1,55 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "storage/rowset/segment_v2/data_decoder.h"
+
+namespace starrocks::segment_v2 {
+
+std::unique_ptr<DataDecoder> get_data_decoder(EncodingTypePB encoding) {
```

Comment:
Can we just return a pointer to a global instance, without creating a DataDecoder in each call?

Reply:
> Can we just return a pointer to a global instance, without creating a DataDecoder in each call?

fix in https://github.com/StarRocks/starrocks/pull/1597/commits/7ebb8723c2037b1e4bf72e9112deb4d92ba1c262, PTAL